### PR TITLE
Implement secrets polling period flag.

### PR DESCRIPTION
**Pull request by BearLemma**

### CODE REVIEW

File: server/src/opt.rs [hunk 1]
```diff
@@ -63,6 +63,10 @@ pub struct Opt {
     #[structopt(long, env = "CHISEL_SECRET_LOCATION")]
     pub chisel_secret_location: Option<String>,
 
+    /// Sets secrets polling period in seconds (can be float).
+    #[structopt(long, default_value = "1")]
```
Comment (haaawk) [inline server/src/opt.rs:67]: Maybe make this millis in case we ever want to poll more often than a second?

Comment (honzasp) [reply server/src/opt.rs:67]: In that case, perhaps this should be a float?

File: server/src/server.rs [hunk 2]
```diff
@@ -278,7 +278,10 @@ async fn refresh_secrets(server: Arc<Server>) -> Result<()> {
         } else {
             last_try_was_failure = false;
         }
-        tokio::time::sleep(Duration::from_millis(1000)).await;
+        tokio::time::sleep(Duration::from_millis(
+            (1000.0 * server.opt.secrets_polling_period_s) as u64,
+        ))
+        .await;
```
Comment (honzasp) [inline server/src/server.rs:284]: I would use `Duration::from_secs_f32(server.opt.secrets_polling_period_s)`

**Review by haaawk — APPROVED** [review]

**Comment by haaawk** [comment]
Build fixed here https://github.com/chiselstrike/chiselstrike/pull/1883

**Pull request closed without merging**